src/coprocessor/codec/mysql/duration.rs

```diff
@@ -32,13 +37,226 @@ fn check_dur(dur: &StdDuration) -> Result<()> {
     Ok(())
 }
 
-fn tm_to_secs(t: Tm) -> u64 {
-    t.tm_hour as u64 * SECS_PER_HOUR + t.tm_min as u64 * SECS_PER_MINUTE + t.tm_sec as u64
+#[inline]
+fn check_hour(hour: u32) -> Result<u32> {
```

Comment:
Looks like these function can be put inside parser as well?

Reply:
Functions like `check_hour` are used not only in parser but also in other methods to build a `Duration` in my next PR. So I would rather keep it outside from the `parser` module. 😆